lib/resources/nginx.rb

```diff
@@ -0,0 +1,108 @@
+# encoding: utf-8
+# copyright: 2017
+# author: Aaron Lippold, lippold@gmail.com
+# author: Rony Xavier, rx294@gmail.com
+# license: All rights reserved
+
+require 'pathname'
+
+module Inspec::Resources
+  class Nginx < Inspec.resource(1)
+    name 'nginx'
+    desc 'Use the nginx InSpec audit resource to test information about your NGINX instance.'
+    example "
+      describe nginx.pid_file do
+        its('owner') { should be 'root' }
+      end
+      describe nginx.version do
+        it { should_be >= 1.0.0 }
+      end
+      describe nginx do
+        its('group') { should be 'root' }
+      end
+    "
+
+    def initialize
+      return skip_resource 'The `nginx` resource is not yet tested on your OS.' if inspec.os.windows?
+
+      cmd = inspec.command('nginx -V 2>&1')
+      if !cmd.exit_status.zero?
+        return skip_resource 'Error using the command nginx -V'
+      end
+      @data = cmd.stdout
+      @params = {}
+      read_content
+    end
+
+    # @todo RFC: which of these methods should be private and or not exposed?
+    #
+
+    def method_missing(name)
+      # ensure params are loaded
+      read_content
+      # extract values
+      @params[name.to_s] unless @params.nil?
+    end
+
+    def read_content
```

Comment:
If there's a chance this could get called more than once for a given resource, we need to avoid reparsing the data each time.

Something like:

```ruby
return unless @params.empty?
parse_config
...
```